error/s2n_errno.h

```diff
@@ -297,6 +297,8 @@ extern __thread const char *s2n_debug_str;
 #endif /* CBMC */
 
 #define S2N_IMPLIES(a, b) (!(a) || (b))
+/* If and only if (iff) is a biconditional logical connective between statements a and b. */
+#define S2N_IFF(a, b) (S2N_IMPLIES(a, b) && S2N_IMPLIES(b, a))
```

Comment:
Is it better to define it this way, or just 
```
S2N_IFF(a, b) (!!(a) == !!(b))
```
I recall that Mark had some issues with the second way, but it might be clearer?

Reply:
I'd prefer to keep it in term of implies, since we had problems with it in MQTT. @camshaft any preferences here?

Reply:
That's a good point. It's probably better to only use macro arguments once, in case there's more expensive calls that get duplicated.

Reply:
You can add a comment explaining it in terms of imples as well

Reply:
Implemented as suggested.